hack/test-integration.sh

```diff
@@ -69,6 +69,7 @@ os::log::start_system_logger
 function exectest() {
 	echo "Running $1..."
 
+	export TEST_ETCD_DIR=$(mktemp ${TMPDIR:-/tmp}/etcd-XXXXXXXX)
```

Comment:
we're also trying to leave the dir in place in failure cases, which means we need unique dirs